lnt/server/ui/views.py

```diff
@@ -1196,15 +1193,16 @@ def trace_name(name, test_name, field_name):
 
             is_multisample = (len(values) > 1)
 
-            aggregation_fn = min
-            if switch_min_mean_local:
-                aggregation_fn = lnt.util.stats.agg_mean
-            if field.bigger_is_better:
-                aggregation_fn = max
+            fn_name = options.get('aggregation_function') or ('max' if field.bigger_is_better else 'min')
+            aggregation_fn = determine_aggregation_function(fn_name)
+            agg_value = aggregation_fn(values)
+
+            # When aggregating multiple samples, it becomes unclear which sample to use for
+            # associated data like the run date, the order, etc. Use the index of the closest
+            # value in all the samples.
+            closest_value = sorted(values, key=lambda val: abs(val - agg_value))[0]
```

Comment:
This is the change I called out about reimplementing how we select related data associated to the aggregated point. Previously, for `min` we'd use the actual data point that we extracted the minimum value from. For `mean` we would always pick the first point's data (see how `agg_mean` returned `0` for the index).

Now, we select the point closest to the value we selected. I can't really imagine a case where this would give us an incorrect answer.

Reply:
I'll merge assuming you're happy with this explanation. If not, let's talk and I can address that in another PR.

Reply:
Ah interesting, your explanation makes sense. The way that we use aggregation on cc-perf.igalia.com is that we use `--exec-multisample=3` to get multiple samples in a single run.

But I didn't consider that you might end up with samples over multiple runs if you use `--merge append`. No problems with the fix here, thanks for talking it through.